### DISCUSSION TITLE
Why @nrwl/js:lib does not collect dependencies?

### DISCUSSION THREAD

**Discussion by ziacik** [nrwl/nx#8569] [General]
Deciding whether I should use @nrwl/js or @nrwl/node, I noticed when I build @nrwl/node lib which has been generated as buildable and publishable, nx collects all dependencies the library uses and puts them into generated package.json. @Nrwl/js library builder does not do that. Is it because it is not publishable? And if so, should i choose @nrwl/node if I want to publish the library?

**Comment by vadistic** [comment]
Adding to this question - there is also `@nrwl/web` - that seems to have capabilities of `@nrwl/node` and more with those webpack + rollup builders.

Which is currently a recommended one for shared libs? Which one for node only?

**Comment by fider-apptimia** [comment]
#frustration
Some of tutorials just not working and some docs seems also to be outdated.
